stock_lock_lot/models/stock_production_lot.py

```diff
@@ -43,12 +43,15 @@ def _get_locked_value(self):
     @api.one
     @api.onchange('product_id')
     def onchange_product_id(self):
-        '''Instruct the client to lock/unlock a lot on product change'''
+        """Instruct the client to lock/unlock a lot on product change"""
         self.locked = self._get_product_locked(self.product_id)
 
     @api.multi
     def button_lock(self):
-        '''Lock the lot if the reservations allow it'''
+        """"Lock the lot if the reservations and permissions allow it"""
```

Comment:
We tried setting the perms client-side, but then they could be bypassed RPC.
We tried setting them at ORM level, but then the field was invisible to users who had no right to change it.
So as a last resort we enforced it this way.